### PR TITLE
Fix/fix shared viewmodel

### PR DESCRIPTION
**Pull request by ted3x**
Created BaseFragment abstract class, where I handle all network connectivity and opening detailed movie.
Now MovieAdapter uses lambda expression instead of interface with one method.
App doesn't use shared view models, since it's useless.
now Fragments can change UI based on connectivity.
ConnectivityReceiver now uses lambda as well instead of interface.

### CODE REVIEW

File: app/src/main/java/com/c0d3in3/movieapp/BaseFragment.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    private fun setErrorMessage(isInternetAvailable: Boolean, errorMessage : String){
+        (activity as MovieActivity).setErrorMessage(isInternetAvailable, errorMessage)
+    }
+
+    protected fun openDetailedMovie(movieId: Int?) {
+        val bundle = Bundle()
+        if (movieId != null) bundle.putInt("movieId", movieId)
+        navController.navigate(R.id.action_moviesDashboardFragment_to_movieDetailFragment,bundle)
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/movieapp/BaseFragment.kt:71]: remake this bundle like bundleOf

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt [hunk 2]
```diff
@@ -18,13 +19,17 @@ import kotlinx.android.synthetic.main.fragment_movie_detail.*
...
-    private val viewModel by activityViewModels<MoviesViewModel>()
+    private lateinit var viewModel : MovieDetailViewModel
     private lateinit var navController: NavController
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+        activity?.let{
+            viewModel = ViewModelProvider(it)[MovieDetailViewModel::class.java]
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt:30]: this should be in onCreate() method

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt [hunk 3]
```diff
@@ -18,13 +19,17 @@ import kotlinx.android.synthetic.main.fragment_movie_detail.*
...
+        }
+        viewModel.loadMovie(requireArguments().getInt("movieId"))
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt:32]: this should be in onViewCreated() method

**Review by Edna001 — CHANGES_REQUESTED** [review]

**Review by DavitSoitashviliSpace — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/c0d3in3/movieapp/BaseFragment.kt (hunk 1 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/BaseFragment.kt:71]: Changed

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt (hunk 2 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt:30]: Changed

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt (hunk 3 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailFragment.kt:32]: Changed

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import retrofit2.Response
+
+class MovieDetailViewModel : ViewModel() {
+    val selectedMovie  = MutableLiveData<Movie>()
+    private val repository = MovieRepositoryImpl()
+    var isFavoriteMovie = MutableLiveData<Boolean>()
+
+    fun loadMovie(movieId: Int){
+        viewModelScope.launch{
+            withContext(Dispatchers.IO){
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt:23]: Changed

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                        }
+                    })
+                }
+            }
+        }
+    }
+
+    private fun checkMovieForFavorite(){
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt:46]: Changed

File: app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                isFavoriteMovie.postValue(repository.getMovieById(selectedMovie.value?.id!!) != null)
+            }
+        }
+    }
+
+    fun handleFavoriteControl(){
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                selectedMovie.value?.let {
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/ui/movie_detail_screen/MovieDetailViewModel.kt:55]: Changed

**Review by DavitSoitashviliSpace — APPROVED** [review]

**Review by Edna001 — APPROVED** [review]

**Pull request merged**